validator/serviceusage_service.rego

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package templates.gcp.GCPServiceUsageConstraintV1
+
+import data.validator.gcp.lib as lib
+
+deny[{
+	"msg": message,
+	"details": metadata,
+}] {
+	constraint := input.constraint
+	lib.get_constraint_params(constraint, params)
+	asset := input.asset
+	asset.asset_type == "serviceusage.googleapis.com/Service"
+
+	service_usage := asset.resource.data
+	parent := service_usage.parent
+	state := service_usage.state
+	service := extract_service(service_usage.name)
+
+	mode := lib.get_default(params, "mode", "allow")
+
+	state == "ENABLED"
+	matches_found = [m | m = params.services[_]; glob.match(m, [], service)]
```

Comment:
Does this need to be a glob or should it be a literal match?

Reply:
@morgante are you suggesting to use: re_match ? I think we probably want a literal match, but don't understand the side effects of using glob

Reply:
I mean just `m == service`.

Reply:
ah yea that makes sense, I don't really see regex being useful for googleapi services